include/nvexec/stream/schedule_from.cuh

```diff
@@ -57,8 +110,7 @@ namespace nvexec::STDEXEC_STREAM_DETAIL_NS {
                     self.operation_state_.propagate_completion_signal(Tag2(), std::move(tas)...);
                   },
                   tpl);
-              },
-              *storage);
+              }, *storage, index);
```

Comment:
Why is it necessary to pass the index here? Just curious.

Reply:
The variant might be constructed on the device asynchronously:

```cpp
if constexpr (construct_on_device) {
  kernel<Tag, storage_t, __decay_t<As>...><<<1, 1, 0, stream>>>(storage, as...);

  if (cudaError_t status = STDEXEC_DBG_ERR(cudaPeekAtLastError());
      status != cudaSuccess) {
    self.operation_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
    return;
  }
} 
```

Without passing index, `visit` reads the index from the storage object, leading to a data transfer, which is not needed. Besides that, following kernel will have to prefetch data back on device. Since we know the index in advance from the `decayed_tuple<Tag, As...>` type, we can avoid this.